drivers/watchdog/wdt_andes_atcwdt200.c

```diff
@@ -0,0 +1,371 @@
+#define REG_WREN		0x18
+#define REG_STATUS		0x1c
+
+#define WDT_BASE		(((const struct wdt_atcwdt200_config *)(dev)->config)->base)
+#define WDT_CTRL(dev)		(WDT_BASE + REG_CTRL)
+#define WDT_RESTAR(dev)		(WDT_BASE + REG_RESTAR)
+#define WDT_WREN(dev)		(WDT_BASE + REG_WREN)
+#define WDT_STATUS(dev)		(WDT_BASE + REG_STATUS)
+
+/* Atcwdt200 magic number */
+/* 0x10 Control Register */
+
+#define WDT_CTRL_RSTTIME_POW_2_7	0x000
+#define WDT_CTRL_RSTTIME_POW_2_8	0x100
+#define WDT_CTRL_RSTTIME_POW_2_9	0x200
+#define WDT_CTRL_RSTTIME_POW_2_10	0x300
+#define WDT_CTRL_RSTTIME_POW_2_11	0x400
+#define WDT_CTRL_RSTTIME_POW_2_12	0x500
+#define WDT_CTRL_RSTTIME_POW_2_13	0x600
+#define WDT_CTRL_RSTTIME_POW_2_14	0x700
+
+#define WDT_CTRL_INTTIME_POW_2_6	0x000
+#define WDT_CTRL_INTTIME_POW_2_8	0x010
+#define WDT_CTRL_INTTIME_POW_2_10	0x020
+#define WDT_CTRL_INTTIME_POW_2_11	0x030
+#define WDT_CTRL_INTTIME_POW_2_12	0x040
+#define WDT_CTRL_INTTIME_POW_2_13	0x050
+#define WDT_CTRL_INTTIME_POW_2_14	0x060
+#define WDT_CTRL_INTTIME_POW_2_15	0x070
+#define WDT_CTRL_INTTIME_POW_2_17	0x080
+#define WDT_CTRL_INTTIME_POW_2_19	0x090
+#define WDT_CTRL_INTTIME_POW_2_21	0x0A0
+#define WDT_CTRL_INTTIME_POW_2_23	0x0B0
+#define WDT_CTRL_INTTIME_POW_2_25	0x0C0
+#define WDT_CTRL_INTTIME_POW_2_27	0x0D0
+#define WDT_CTRL_INTTIME_POW_2_29	0x0E0
+#define WDT_CTRL_INTTIME_POW_2_31	0x0F0
+
+#define WDT_CTRL_RSTEN			0x8
+#define WDT_CTRL_INTEN			0x4
+#define WDT_CTRL_APBCLK			0x2
+#define WDT_CTRL_EXTCLK			0x0
+#define WDT_CTRL_EN			0x1
+
+/* Magic Number for Restart Register */
+#define WDT_RESTART_NUM                 0xcafe
+
+/* Magic Number for Write Enable Register */
+#define WDT_WREN_NUM                    0x5aa5
+
+/* 0x1C Status Register */
+#define WDT_ST_INTEXPIRED               0x1
+#define WDT_ST_INTEXPIRED_CLR           0x1
+
+/*
+ * SMU(System Management Unit) Registers for hwinfo driver
+ */
+
+/* Register offset*/
+#define SMU_RESET_WRSR			0x10
+#define SMU_RESET_REGHI			0x60
+#define SMU_RESET_REGLO			0x50
+#define SMU_CMD				0x14
+
+#define SMU_RESET_CMD			0x3c
+
+#define WDOGCFG_PERIOD_MIN		(0x1 << 7)
+#define WDOGCFG_PERIOD_MAX		(0x1 << 14)
+#define EXT_CLOCK_FREQ			(0x1 << 15)
+
+static const struct device *const syscon_dev =
+				DEVICE_DT_GET(DT_NODELABEL(syscon));
+static const struct device *const counter_dev =
+				DEVICE_DT_GET(DT_NODELABEL(pit0));
+
+struct counter_alarm_cfg alarm_cfg;
+
+struct wdt_atcwdt200_config {
+	uintptr_t base;
+};
+
+struct wdt_atcwdt200_dev_data {
+	bool timeout_valid;
+	counter_alarm_callback_t counter_callback;
+	struct k_spinlock lock;
+};
+
+static int wdt_atcwdt200_disable(const struct device *dev);
+
+int __aligned(16) __attribute__((naked)) nmi_handler(void) {
+
+	const struct device *dev = DEVICE_DT_INST_GET(0);
+
+	sys_write32(WDT_WREN_NUM, WDT_WREN(dev));
+	sys_write32(~(WDT_CTRL_RSTEN | WDT_CTRL_EN), WDT_CTRL(dev));
+
+	syscon_write_reg(syscon_dev, SMU_RESET_WRSR, 0x1f);
+	__asm volatile("j entry");
```

Comment:
needs some comments here

Reply:
We have remove this function, because we have confirmed that our wdt ip will do the reg reset automatically.